### PR TITLE
Remove Vendor folder

### PR DESCRIPTION
**Pull request by edwardecook**
It was mentioned in a previous PR that it might be nice to remove the vendor folder, https://github.com/kubernetes-sigs/go-open-service-broker-client/pull/149#pullrequestreview-404492514.

This also includes bumping the version of golang used in travis to one that supports go modules.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: go
 go_import_path: github.com/kubernetes-sigs/go-open-service-broker-client
 go:
-  - 1.7.4
```
Comment (MHBauer) [inline .travis.yml:4]: 😬

**Comment by MHBauer** [comment]
/lgtm
/approve
I'm fine with it, others might not be.
/hold

**Comment by MHBauer** [comment]
this also brings up the existence of klog v2 which should probably make an issue to move to. Or otherwise evaluating logging in general.

**Comment by edwardecook** [comment]
@MHBauer yeah I agree that moving to klog v2 makes sense, I held off on doing that as part of this PR as I didn't want to muddle things too much.

**Comment by mszostok** [comment]
For such a small lib I'm also fine with removing the `vendor` folder.
/lgtm

As a side note, in the Service Catalog repo we still commit the vendor folder as k8s does. By doing so, we have faster and more stable builds but the deps in the svcat project are much bigger and we also have 13 pipelines which we want to keep below 30min.

**Comment by MHBauer** [comment]
yea, given it's klog, and people are gonna have it as a direct dependency most likely anyway, I'm good with it.

service-catalog should keep the vendor folder for sure. Lots of people expect it to have one like k8s does.

**Pull request merged**